apps/web/components/dashboard/sidebar-pinboard.tsx

```diff
@@ -74,27 +145,173 @@ const PinboardListButton = forwardRef<
+  const [emojiPopoverOpen, setEmojiPopoverOpen] = useState(false);
+  const { toast } = useToast();
+
+  const updateListEntry = async (updateData: {
+    name?: string;
+    favorite?: boolean;
+    icon?: string;
+  }) => {
+    const { error } = await updateList(item.slug, updateData);
+    if (error) {
+      toast({
+        title: "Couldn't update list",
+        description:
+          'An error occurred while updating the name. Please try again later.',
+      });
+    } else {
+      setPopoverOpen(false);
+      setEmojiPopoverOpen(false);
+    }
+  };
+
+  const duplicateListEntry = async () => {
+    const { error } = await createList(listName);
+    if (error) {
+      toast({
+        title: "Couldn't duplicate list",
+        description:
+          'An error occurred while duplicating the list. Please try again later.',
+      });
+    } else {
+      setPopoverOpen(false);
+    }
+  };
+
+  const deleteListEntry = async () => {
+    const { error } = await deleteList(item.slug);
+    if (error) {
+      toast({
+        title: "Couldn't delete list",
+        description:
+          'An error occurred while deleting the list. Please try again later.',
+      });
+    } else {
+      setPopoverOpen(false);
+    }
+  };
 
   return (
-    <div ref={ref} className="flex justify-center space-y-1" {...rest}>
+    <div
+      ref={ref}
+      className="group/item flex justify-center space-y-1"
+      {...rest}
+    >
       <Button
         asChild
         variant="ghost"
         className={cn(
           'w-full',
           item.slug === segment ? 'bg-accent' : '',
           isCollapsed()
-            ? 'inline-flex w-auto items-center justify-center px-[10px]'
+            ? 'flex w-auto items-center justify-center px-[10px]'
             : 'inline-block justify-start overflow-hidden overflow-ellipsis whitespace-nowrap px-4 py-2'
         )}
       >
-        <Link href={item.href}>
-          <span className={cn('text-base', isCollapsed() ? 'pr-0' : 'pr-1')}>
-            {item.icon}
-          </span>
-          {!isCollapsed() && <span>{item.name}</span>}
-        </Link>
+        <div className="flex w-auto flex-row items-center justify-between">
+          <div className="flex flex-row items-center">
+            <Popover open={emojiPopoverOpen} onOpenChange={setEmojiPopoverOpen}>
+              <PopoverTrigger>
+                <span
+                  className={cn(
+                    'text-center text-base',
+                    isCollapsed() ? 'pr-0' : 'pr-2'
+                  )}
+                >
+                  {item.icon}
+                </span>
+              </PopoverTrigger>
+              <PopoverContent side="right" className="w-fit">
+                <EmojiPicker
+                  style={{ border: 'none' }}
+                  onEmojiClick={async obj => {
+                    await updateListEntry({ icon: obj.emoji });
+                  }}
+                />
```

Comment:
Can we add a slightly different background color on hover so that the user can anticipate that clicking on the emoji opens the picker?